### DISCUSSION TITLE
How to Develop Extensions for Microsoft Edge on Android?

### DISCUSSION THREAD

**Discussion by nanqic** [microsoft/MicrosoftEdge-Extensions#195] [Q&A]
Hi everyone,

I'm interested in developing extensions for Microsoft Edge on Android, but I'm not quite sure how to get started. Could anyone help me with the following questions?

1. Development Environment: What tools and software do I need to begin development? Is there a specific version of Edge I should use?

1. Extension Structure: Is the extension structure for Android similar to that of the desktop version? Are there any specific file structures or configuration requirements?

1. API Support: What WebExtension APIs are supported on Android? Are there any limitations or unsupported features I should be aware of?

1. Testing and Debugging: What are the best practices for testing extensions on Android devices?

1. Publishing Process: How do I submit my extension to the Microsoft Edge Add-ons site for review?

Thank you so much for your help! I look forward to your responses.

**Comment by ManikanthMSFT** [comment]
Hi @nanqic, thank you for contacting us. Could you please confirm us are you willing to create extension for Edge Android or on Android?

Also, you can refer to our [documentation ](https://learn.microsoft.com/en-us/microsoft-edge/develop-web-microsoft-edge)which will help you in building the edge extension.
